ts/features/lollipop/index.ts

```diff
@@ -0,0 +1,57 @@
+import { PublicKey } from "@pagopa/io-react-native-crypto";
+import URLParse from "url-parse";
+import { SignatureAlgorithm } from "../../utils/httpSignature/types/SignatureAlgorithms";
+import { SignatureComponents } from "../../utils/httpSignature/types/SignatureComponents";
+
+export type LollipopConfig = {
+  nonce: string;
+  customSignatures?: ReadonlyArray<string>;
+};
+
+/**
+ * Utility function to forge the `SignatureComponents` based on the provided inputs.
+ */
+export function forgeSignatureComponents(
+  method: string,
+  inputUrl: URLParse,
+  originalUrl: string
+): SignatureComponents {
+  return {
+    method,
+    authority: inputUrl.hostname,
+    path: inputUrl.pathname,
+    requestTarget: originalUrl,
+    scheme: inputUrl.protocol,
+    targetUri: `${inputUrl.protocol}://${inputUrl.hostname}/${originalUrl}`
+  };
+}
+
+/**
+ * Returns the http-signature algorithm used to sign the signature base specified by
+ * the signature-input header.
+ */
+export function getSignAlgorithm(publicKey: PublicKey): SignatureAlgorithm {
+  return publicKey.kty === "EC" ? "ecdsa-p256-sha256" : "rsa-pss-sha256";
+}
+
+/**
+ * Result type of promises returnd by `chainSignPromises`
```

Comment:
```suggestion
```

Reply:
avoid to put this kind of implementation details into comments since they may change